src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -122,13 +122,13 @@ public static bool IsPathRooted(ReadOnlySpan<char> path)
         public static string? GetPathRoot(string? path)
         {
             if (PathInternal.IsEffectivelyEmpty(path)) return null;
-
             return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString : string.Empty;
         }
 
         public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
         {
-            return PathInternal.IsEffectivelyEmpty(path) && IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString.AsSpan() : ReadOnlySpan<char>.Empty;
+            if (PathInternal.IsEffectivelyEmpty(path)) return null;
```

Comment:
`null` and `Empty` are identical for Spans. I believe that this just makes the code a bit less efficient and readable, without functional difference. Am I missing something?

Reply:
I am returning `ReadOnlySpan<char>.Empty` now.
There is a functional difference though.

Reply:
Ok, I see the functional difference now. I believe that the check for `IsEffectivelyEmpty` should not be needed at all.